### PR TITLE
RCT unpacker plus DQM 76X

### PR DESCRIPTION
**Pull request by nsmith-**
This is a 76X copy of #10381, without the `EventFilter/RctRawToDigi` history, which has already been merged in 76X (PR #10381)

### CODE REVIEW

**Comment by jingyucms** [comment]
@deguio Can you please test and review? The 74x version of this PR are running fine in dev machine.

**Comment by deguio** [comment]
please test

File: DQM/L1TMonitor/src/L1TRCT.cc [hunk 1]
```diff
@@ -176,35 +213,63 @@ void L1TRCT::analyze(const Event & e, const EventSetup & c)
...
+    doHdLayer2 = false;
+  }
+
+  if (!emLayer2.isValid()) {
+    edm::LogInfo("DataNotFound") << "can't find L1CaloEmCollection - Layer2";
+    doEmLayer2 = false;
+  }
+
+
+  std::cout<<doHd<<doEm<<doHdLayer2<<doEmLayer2<<std::endl;
```
Comment (deguio) [inline DQM/L1TMonitor/src/L1TRCT.cc]: please remove cout

**Comment by deguio** [comment]
-1
apart from the cout to be removed, the PR looks ok and can go in.
rejecting for now, but feel free to update this branch and the PR will be reconsidered.

**Comment by deguio** [comment]
please test

**Comment by deguio** [comment]
+1

**Comment by nsmith-** [comment]
@jingyucms do you know why the `caloStage1Digis` are not created in these tests?

**Comment by jingyucms** [comment]
@nsmith caloStage1Digis is created. It fail because L1TdeRCT is looking for a std::vector format product but caloStage1Digis is actually a BxVector. You need a convertor to convert BxVector to std::vector to automatically feed the plots. And this is what I am doing.

**Comment by nsmith-** [comment]
@jingyucms Are you sure?
Take the 1003.0 step2 RelVal for example, download the autogenerated config that is here:

https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10658/7305/runTheMatrix-results/1003.0_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM/step2_RAW2DIGI_L1Reco_RECO_EI_DQM.py

And in the directory, try:
```
edmConfigDump step2_RAW2DIGI_L1Reco_RECO_EI_DQM.py > process.dump
grep 'caloStage1Digis' process.dump
grep 'L1TRawToDigi' process.dump
```
The only things I see when I do this are the L1TdeRCT modules trying to read in these digis.
`L1TRawToDigi` is the unpacker that creates the digis, and I won't expect it to work on a data file from 2012 as there was no stage 1 in 2012!

**Comment by danduggan** [comment]
-1
still needs another commit before ready to go in

**Comment by jingyucms** [comment]
@nsmith- can you run caloStage1Digis in l1TdeRCTSeq and l1TRCTSeq? ( L1TMonitor_cff.py and L1TEmulatorMonitor_cff.py)

**Comment by nsmith-** [comment]
@jingyucms I think that unpacker steps should not belong in DQM-specific workflows, esp. when the unpackers for Stage-1 must certainly be in `StandardSequences` as we transition away from having a GCT.  Perhaps @mulhearn can elaborate.
@danduggan Can you expand your comment?  As far as I understand, this PR is being held back due to failing a RelVal that is impossible to pass.  Is it not possible to update the RelVals to reflect the fact that there are new FEDs, new unpackers, and therefore new event content for DQM in 2015 runs?

**Comment by jingyucms** [comment]
@nsmith-  I think the falling RelVal can pass if you add the unpacker into your RCT sequence. Can you try?

**Comment by jingyucms** [comment]
@nsmith- And add it to standard sequences will work also.

**Comment by danduggan** [comment]
please test

**Comment by nsmith-** [comment]
Superceded by #10882

**Pull request closed without merging**